### PR TITLE
New Block: Editor Promo Footer

### PR DESCRIPTION
**Pull request by dmsnell**
Because I'm tired of manually writing this at the end of all my posts
and I would love for other people to be spreading news about the editor
I created a block to append a basic advert intended as a post footer.

editorpromoblock

### CODE REVIEW

**Comment by mtias** [comment]
I like this little block :)

It would also show how similar ones can be created.

File: blocks/library/gutenvert/index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	title: __( 'Editor Promo' ),
+
+	icon: 'megaphone',
+
+	category: 'common',
+
+	save() {
+		return (
+			<div>
+				<hr className="wp-block-separator" />
```
Comment (iseulde) [inline blocks/library/gutenvert/index.js]: Do you think we should leave it up to the author to insert this or not? Or maybe use `<footer>` and style that (differently) together with the styles below?

File: blocks/library/gutenvert/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+				<p style={ { color: '#9e9da4', fontSize: '14px', textAlign: 'center' } }>
+					This post brought to you by the new
+					{ ' ' }<a href="https://wordpress.org/plugins/gutenberg/">Gutenberg editor</a> for
```
Comment (iseulde) [inline blocks/library/gutenvert/index.js]: Huh, is that needed?

**Comment by iseulde** [comment]
This block makes me want to add [sparkles](https://codepen.io/simeydotme/pen/jgcvi) to it.

File: blocks/library/gutenvert/index.js (hunk 2 above)
Comment (dmsnell) [reply blocks/library/gutenvert/index.js]: it's not _needed_ but I wanted to wrap the code before the `<a>` and it was awkward no matter how I diced it. could have also inserted an entity to encode the space but I felt like the literal JS space was more clear. thoughts? (without it we lose the space between `new` and the link text)

File: blocks/library/gutenvert/index.js (hunk 1 above)
Comment (dmsnell) [reply blocks/library/gutenvert/index.js]: `<footer>` sounds good - I'll have to check on styling but I like the idea - thanks!

**Comment by karmatosed** [comment]
Could we make this a credits block over say Gutenberg? My thoughts in this are, a credits block would be really useful for users. I think having this just as a Gutenberg promotional block isn't as useful. I feel having a credits block that is editable could be really great.

**Comment by iseulde** [comment]
What would a credits block look like? How would it be different form a separator and an editable paragraph?  If we just want someone to have the option to spread the word, maybe we can add it when a new post is created, where you can of course delete it or overwrite it. This is kind of like the "Sent from my iPhone" that Apple adds to new emails (unless you disable it because it's not like they need it).

**Comment by karmatosed** [comment]
@iseulde I think we could ship this if it was editable as the credits block.

**Comment by dmsnell** [comment]
@karmatosed definitely open to the idea but I don't think I understand what you mean when you say "credits block" - could you help by clarifying that meaning, how you think people would want to use it, and how you think someone might want to change it?

**Comment by karmatosed** [comment]
I think users would want to edit it. I feel it's limiting having just as not editable. A possible note could be: call for action or for having 'made by', maybe a sponsored link.

What I am thinking of would be something like this:

- It looks as does now.
- It's called 'credits' and loads the content we have now as default but is editable.

Bonus points if this can be saved as a block globally depending on text - this feels future :)

This way people can use it and edit their own message if they want. I'm thinking just having a promo block isn't as good for the project as having a credit block people can reuse.

**Comment by dmsnell** [comment]
> Bonus points if this can be saved as a block globally depending on text - this feels future :)

That's my feeling on it. My entire motivation for writing it was to have a sort of macro to insert the same static content each time. Making it editable would be easy I think but also it's complexity I didn't see as warranted.

In the meantime I can try and update this PR with it editable and having the Gutenberg defaults and we can play with it more. Thanks @karmatosed!

**Comment by karmatosed** [comment]
@dmsnell would you still be up for iterating on this?

**Comment by dmsnell** [comment]
sure @karmatosed - just have been distracted by other work recently, plus the scope broadening requested here. I'll see if I can hook around to it within the next week

**Comment by dmsnell** [comment]
I'm closing this since it's not really relevant after the introduction of saved templates and locked blocks.

**Comment by mtias** [comment]
@dmsnell what do you think of adding this to one of our docs? The simplicity and non-editable nature of it seems convenient for an example doc.

**Pull request closed without merging**